server/backend/db_backend.go

```diff
@@ -96,7 +96,7 @@ func (self *MongoBackend) parseBlock(block *types.Block) *models.Block {
 		ParentHash: block.ParentHash().Hex(),
 		TxHash:     block.Header().TxHash.Hex(),
 		GasUsed:    strconv.Itoa(int(block.Header().GasUsed)),
-		Nonce:      uint64(block.Nonce()),
+		Nonce:      block.Nonce(),
```

Comment:
It looks like the field and the method return are both already `uint64`, so what does this fix exactly? Isn't it just removing a redundant type conversion?

Reply:
yep just removing redundant type conversion

Reply:
How does that fix the error? Isn't the code behaving the same either way?

Reply:
![image](https://user-images.githubusercontent.com/40002763/59630159-57058680-914d-11e9-99fe-47fcf98ba88e.png)

Reply:
`0` is not the correct value for the nonce of that block. It is `0xffffffffffffffff`, not `0x0000000000000000`.

Reply:
https://rpc.gochain.io/x/eth_getBlockByNumber/6869909

Reply:
I don't think it is valid to pass a max uint64 to javascript, unless we are handling it carefully with a big number library. But regardless, we don't even really want those values on the front end anyways since the field is only relevant during a vote, and only has two possible values.
https://github.com/gochain-io/explorer/issues/233

Reply:
so do you want to see hex value there?

Reply:
> so do you want to see hex value there?

No. The linked issue goes in to detail about how I thought it should be displayed. The interpretation of the hex values can probably be done on the backend to avoid this whole mess, and a simpler `bool` or `*bool` can be passed to the front end.

Reply:
so maybe we could merge this to fix the error with non-loading block and I(or Murat) will work on #233 separately?

Reply:
I don't think that this fixes the error. I would bet that the code is compiling exactly the same with or without the superfluous type conversion, and something else is going on that prevented the crash. IMO better intermediate step would be to remove the nonce from the front end entirely, since it's not useful to the users, especially if it's always 0.

Reply:
it crashes on backend, after this fix it doesn't crash (as on screenshot above), feel free to test yourself

Reply:
> BSON has no uint64 type, and value is too large to fit correctly in an int64

Doesn't this error suggest that the nonce cannot be stored in mongo as a uint64?

Reply:
> it crashes on backend, after this fix it doesn't crash (as on screenshot above), feel free to test yourself

There must be something else going on. The conversion doesn't do anything.

Reply:
interesting )

Reply:
let me finish migration to gce and then I'll check 233